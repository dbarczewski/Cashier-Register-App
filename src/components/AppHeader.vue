<template>
  <div
    class="
      w-screen
      bg-background
      text-gray-300
      justify-between
      flex
      items-center
    "
  >
    <nav class="ml-16 flex flex-row space-x-16 my-2">
      <div><LockedIcon /></div>
      <div><MenuIcon /></div>
      <div><MenuArrowIcon /></div>
    </nav>
    <div class="flex flex-row items-center">
      <div
        class="
          bg-gray-100
          h-full
          border-t-8 border-background
          flex flex-row
          relative
          justify-center
          items-center
          pt-4
        "
      >
        <avatar-icon class="m-2 p-6" />
        <div class="text-gray-800 mr-16">
          <div>{{ content.sellingLabel }}</div>
          <div>012</div>
        </div>
        <button class="absolute right-2 top-0">x</button>
      </div>
      <div class="px-12">
        <PlusIcon />
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import LockedIcon from "./icons/LockedIcon.vue";
import MenuIcon from "./icons/MenuIcon.vue";
import MenuArrowIcon from "./icons/MenuArrowIcon.vue";
import PlusIcon from "./icons/PlusIcon.vue";
import AvatarIcon from "./icons/AvatarIcon.vue";
import Content from "./../../content.json";

export default defineComponent({
  name: "Header",
  components: {
    LockedIcon,
    MenuIcon,
    MenuArrowIcon,
    PlusIcon,
    AvatarIcon,
  },
  data() {
    return {
      content: Content,
    };
  },
});
</script>
