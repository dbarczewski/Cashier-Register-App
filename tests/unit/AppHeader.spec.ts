import { shallowMount } from "@vue/test-utils";
import AppHeader from "@/components/AppHeader.vue";

describe("icons/LockedIcon", () => {
  it("renders component correctly", () => {
    const wrapper = shallowMount(AppHeader);
    expect(wrapper.element).toMatchSnapshot();
  });
});
