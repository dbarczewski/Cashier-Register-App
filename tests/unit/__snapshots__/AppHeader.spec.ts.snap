// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons/LockedIcon renders component correctly 1`] = `
<div
  class="w-screen bg-gray-700 text-gray-200 justify-between flex items-center"
>
  <nav
    class="ml-6 flex flex-row space-x-6"
  >
    <div>
      <locked-icon-stub />
    </div>
    <div>
      <menu-icon-stub />
    </div>
    <div>
      <menu-arrow-icon-stub />
    </div>
  </nav>
  <div
    class="flex flex-row items-center"
  >
    <div
      class="
          bg-white
          h-full
          border-t-2 border-gray-700
          flex flex-row
          relative
          pt-2
        "
    >
      <avatar-icon-stub
        class="m-2"
      />
      <div
        class="text-gray-800 mr-12"
      >
        <div>
          Selling
        </div>
        <div>
          012
        </div>
      </div>
      <button
        class="absolute right-2 top-0"
      >
        x
      </button>
    </div>
    <div
      class="px-6"
    >
      <plus-icon-stub />
    </div>
  </div>
</div>
`;
